src/Microsoft.PowerShell.Commands.Utility/commands/utility/AddType.cs

```diff
@@ -598,8 +598,19 @@ protected override void EndProcessing()
 
         // We now ship .Net Core's reference assemblies with PowerShell, so that Add-Type can work
         // in a predictable way and won't be broken when we move to newer version of .NET Core.
-        // The reference assemblies are located at '$PSHOME\ref'.
-        private static readonly string s_netcoreAppRefFolder = PathType.Combine(PathType.GetDirectoryName(Assembly.GetEntryAssembly().Location), "ref");
+        // The reference assemblies are located at '$PSHOME\ref' for pwsh.
+        //
+        // For applications that host PowerShell, the 'ref' folder will be deployed to the 'publish'
+        // folder, not where 'System.Management.Automation.dll' is located. So here we should use
+        // the entry assembly's location to construct the path to the 'ref' folder.
+        // For pwsh, the entry assembly is 'pwsh.dll', so the entry assembly's location is still
+        // $PSHOME.
+        // However, 'Assembly.GetEntryAssembly()' returns null when the managed code is called from
+        // unmanaged code (PowerShell WSMan remoting scenario), so in that case, we continue to use
+        // the location of 'System.Management.Automation.dll'.
```

Comment:
@daxian-dbw Thanks for the great comment! Taking in account that we have very few docs for SDK and hosting I think it makes sense to open new documentation issue for this.

Reply:
As for documentation, I think we should handle the reference assemblies in a way that it just works, and users can use `Add-Type` without needing to consider how it works.
From that perspective, we don't need to surface this in the help docs. It should be internal implementation docs if we want to doc it.